braid/base.c

```diff
@@ -466,6 +633,54 @@ _braid_BaseBufPack(braid_Core          core,
       _braid_VectorBarCopy(u->bar, &ubar_copy);
       _braid_CoreElt(core, barTape) = _braid_TapePush(_braid_CoreElt(core, barTape), ubar_copy);
    }
+
+   if ( _braid_CoreElt(core, delta_correct) )
+   {
+      /*
+       * Delta correction requires sending a full set of basis vectors alongside every state vector.
+       * Using a header prepended to the buffer, we can efficiently pack all of these vectors contiguously
+       * in one message.
+      */
+      braid_Int *header, head_size, tot_size;
+      braid_Byte* data_buf;
+
+      /*
+       * header pointer is offset by one, so
+       * header[-1] stores number of vectors,
+       * header[0] stores size of user vector 
+       * header[i] stores size of ith basis vector 
+       */
+      header = ((braid_Int*)buffer) + 1;
+      header[-1] = u->basis->rank + 1;                   /* total number of vectors */
+      head_size = (header[-1] + 1) * sizeof(braid_Int);  /* header[-1] + size of each vector */
+
+      data_buf = ((braid_Byte*)buffer) + head_size;      /* buffer must have type for legal pointer arithmetic */
+
```

Comment:
Nice job packing all the vectors!

Reply:
Thanks! I think it's pretty slick! I also like the typedef "braid_Byte". It has a nice ring to it :)